### PR TITLE
Upgrade conjur-api-go to latest version (0.12.4)

### LINKED ISSUE #135: Upgrade to latest conjur-api-go release that includes more robust support for JWT authentication
Hi, we are trying to use conjur terraform provider using JWT authentication.  It seems that functionality was added to the conjur-api-go but this provider seems to still be loading version 0.10.1. The latest version at this time is 0.12.4.  If we can update the provider to use the latest version of the conjur-api-go client that would be very helpful to us.

### PR DESCRIPTION
**Pull request by karand1979**
### Desired Outcome
The desired outcome of this pull request is to upgrade to the latest version of the conjur-api-go package and any supporting dependencies.  This will allow our organization to utilize the new environment variables supported by the higher version.

### Implemented Changes
We ran following commands to upgrade the version and ran the ./bin/build to ensure everything built.  Also ran ./bin/test oss
go get github.com/cyberark/conjur-api-go@latest

### Connected Issue/Story

Resolves #135 

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Comment by karand1979** [comment]
@szh can we get this pushed through? It's really important to us and our organization that this gets updated

**Comment by szh** [comment]
@karand1979 I'm bringing it up to my team right now and we'll try to get this merged ASAP.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [0.6.9] - 2024-10-14
+### Changed
+  - Updated direct dependencies (conjur-api-go -> 0.12.4)
```
Comment (szh) [inline CHANGELOG.md:9]:
```suggestion
  - Update conjur-api-go to v0.12.4
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [0.6.9] - 2024-10-14
```
Comment (szh) [inline CHANGELOG.md:7]:
```suggestion
## [0.6.9] - 2024-10-21
```

**Review by szh — COMMENTED** [review]
Can you please address the style changes I've suggested in the CHANGELOG.md file, and also ensure you run `go mod tidy`? Then if you can squash your changes into a single commit and signoff on it (`git commit --amend --signoff`, or see the docs [here](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s)) I should be able to get them included in the next release.

**Comment by karand1979** [comment]
@szh, let me know if it's any better as I'm not sure I did the squash and signoff correctly.  If not I may have to fork it again and do everything in one commit.

**Pull request closed without merging**